agent/functions/guard.go

```diff
@@ -2,25 +2,39 @@ package functions
 
 import (
 	"fmt"
+	"path/filepath"
 	"strings"
 )
 
+// guardPath checks if the given path is safe to access within the current directory context
+// It prevents path traversal attacks and ensures the path is within allowed boundaries
 func guardPath(path string) error {
+	// Basic validation checks
 	if strings.Contains(path, "..") {
 		return fmt.Errorf("path %s contains not allowed '..'", path)
 	}
-
 	if strings.Contains(path, "~") {
 		return fmt.Errorf("path %s contains not allowed '~'", path)
 	}
-
 	if strings.Contains(path, "//") {
 		return fmt.Errorf("path %s contains not allowed '//'", path)
 	}
-
 	if strings.HasPrefix(path, "/") {
 		return fmt.Errorf("path %s starts with '/', not allowed", path)
 	}
 
+	// Clean and normalize the path
+	cleanPath := filepath.Clean(path)
+
+	// Ensure the path is local (doesn't try to escape current directory)
+	if !filepath.IsLocal(cleanPath) {
```

Comment:
from Security_Path_Validator
filepath.IsLocal()は完全な保護を提供しません。絶対パスの検証やボリューム指定（Windowsのドライブレター等）のチェックも必要です。

```suggestion
	// Check for absolute paths and volume specifications
	if filepath.IsAbs(cleanPath) || filepath.VolumeName(cleanPath) != "" {
		return ErrInvalidPath
	}
```